### DISCUSSION TITLE
need help : some images got broken when sequence images are generated from a gif

### DISCUSSION THREAD

**Discussion by meganeroppy** [dlemstra/Magick.NET#1209] [Help]
Hi.

What I want to do is quite simple one that converts a gif into sequence png images, but it doesn't look performing really well.
The result is like one I attached to this post, like some images, that contains pixels that aren't changed from previous frame, are not well converted and get broken.
How I use Magick.NET is this below.

MagickImageCollection collection= new MagickImageCollection(gifPath, MagickFormat.Gif87);
int c = 0;
foreach (var item in images)
{
    var outPath = outDir + "/" +  c.ToString() + ".png";
    item.Write(outPath, MagickFormat.Png);
    
    c++;
}

Is there any way to solve this problem?
Any advise or comment will be appreciated.

Thanks.

![broken images](https://user-images.githubusercontent.com/5607712/176596307-e44b4397-aa3a-4db8-93e6-3b4dbb1cabef.PNG)
![02](https://user-images.githubusercontent.com/5607712/176596331-e88ecbf1-c81b-4f0f-ad90-1cf63882f339.gif)

**Comment by dlemstra** [comment]
I am not sure what you are trying to do. This gif image seems to be optimized and some of the frames contain only the changed pixels. If you want to get each frame as it is shown I would advise you to call `collection.Coalesce()` before you write the frames to the output folder.

**Reply by meganeroppy** [reply]
Thanks for replying dlemstra.

Your advise solved the problem. Thank you so much ;)
In addition I noticed creating MagickImageCollection as apng ( even the original file is gif ) also could make better result.

 Thanks a lot anyway.
